instance/tasks.py

```diff
@@ -39,29 +37,24 @@
 # Tasks #######################################################################
 
 @db_task()
-def provision_instance(instance_pk):
+def spawn_appserver(instance_ref_id, mark_active_on_success=False, num_attempts=1):
     """
-    Run provisioning on an existing instance
-    """
-    logger.info('Retreiving instance: pk=%s', instance_pk)
-    instance = SingleVMOpenEdXInstance.objects.get(pk=instance_pk)
-
-    logger.info('Running provisioning on %s', instance)
-    instance.provision()
-
+    Create a new AppServer for an existing instance.
 
-@db_periodic_task(crontab(minute='*/1'))
-def watch_pr():
+    Optionally mark the new AppServer as active when the provisioning completes.
+    Optionally retry up to 'num_attempts' times
     """
-    Automatically create/update sandboxes for PRs opened by members of the watched
-    organization on the watched repository
-    """
-    team_username_list = get_username_list_from_team(settings.WATCH_ORGANIZATION)
-
-    for username in team_username_list:
-        for pr in get_pr_list_from_username(username, settings.WATCH_FORK):
-            sub_domain = 'pr{number}.sandbox'.format(number=pr.number)
-            instance, created = SingleVMOpenEdXInstance.objects.update_or_create_from_pr(pr, sub_domain)
-            if created:
-                logger.info('New PR found, creating sandbox: %s', pr)
-                provision_instance(instance.pk)
+    for i in range(1, num_attempts + 1):
+        logger.info('Retrieving instance: ID=%s', instance_ref_id)
+        # Fetch the instance inside the loop, in case it has been updated
+        instance = OpenEdXInstance.objects.get(ref_set__pk=instance_ref_id)
+
+        instance.logger.info('Spawning new AppServer on %s, attempt %d of %d', instance, i, num_attempts)
+        result = instance.spawn_appserver()
```

Comment:
@bradenmacdonald Maybe rename `result` to `appserver_id` or `appserver_pk`, and return `None` instead of `False` from `OpenEdXInstance.spawn_appserver` if provisioning fails.